atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -232,6 +232,28 @@ def get_registries(workflow, fallback=NO_FALLBACK):
     return registries_cm
 
 
+def get_docker_registry(workflow, fallback=NO_FALLBACK):
+    try:
+        all_registries = get_config(workflow).conf['registries']
+    except KeyError:
+        if fallback != NO_FALLBACK:
+            return fallback
+        raise
+
+    for registry in all_registries:
+        reguri = RegistryURI(registry.get('url'))
+        if reguri.version == 'v2':
+            regdict = {
+                'url': registry,
+                'insecure': registry.get('insecure', False)
+            }
+            if registry.get('auth'):
+                regdict['secret'] = registry['auth']['cfg_path']
+            return regdict
+
+    return fallback
```

Comment:
We should probably raise an exception here instead. The `fallback` is only meant to be used if the key is not found in reactor config.

Reply:
Fixed.